capstone/src/test/java/com/google/sps/servlets/clubs/EditClubServletTest.java

```diff
@@ -0,0 +1,126 @@
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class EditClubServletTest {
+
+  private final String SAMPLE_CLUB_NAME = "Test Club";
+  private final String SAMPLE_CLUB_DESC_1 = "Test club description";
+  private final String SAMPLE_CLUB_WEB = "www.test-club.com";
+  private final String TEST_EMAIL = "test-email@gmail.com";
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private DatastoreService datastore;
+  private EditClubServlet editClubServlet;
+
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig().setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+          new LocalUserServiceTestConfig(),
+          new LocalBlobstoreServiceTestConfig());
+
+  @Before
+  public void setUp() throws IOException {
+    helper.setUp();
+    MockitoAnnotations.initMocks(this);
+    this.editClubServlet = new EditClubServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doPost_editClubAllValidOfficers() throws ServletException, IOException {
+    prepClubEnv();
+    String newDescription = "new test description";
+    String newWebsite = "new-website.com";
+    String newOfficer = "kakm@google.com";
+    when(request.getParameter(Constants.PROPERTY_NAME)).thenReturn(SAMPLE_CLUB_NAME);
+    when(request.getParameter(Constants.OFFICER_PROP)).thenReturn(TEST_EMAIL + "," + newOfficer);
+    when(request.getParameter(Constants.DESCRIP_PROP)).thenReturn(newDescription);
+    when(request.getParameter(Constants.WEBSITE_PROP)).thenReturn(newWebsite);
+    editClubServlet.doPost(request, response);
+
+    Query query =
+        new Query("Club")
+            .setFilter(
+                new FilterPredicate(
+                    Constants.PROPERTY_NAME, FilterOperator.EQUAL, SAMPLE_CLUB_NAME));
+    Entity clubEntity = datastore.prepare(query).asSingleEntity();
+
+    Assert.assertNotNull(clubEntity);
+    Assert.assertEquals(newDescription, clubEntity.getProperty(Constants.DESCRIP_PROP));
+    Assert.assertEquals(newWebsite, clubEntity.getProperty(Constants.WEBSITE_PROP));
+    Assert.assertEquals(
```

Comment:
nit: it's generally better to assert list size and then individual elements than to check a whole list. This way, an assertion fails, you know which element might be causing issues rather than having to sift through the whole list. So something like:

newOfficers = ImmutableList.copyOf(... clubEntity.getProperty(....));
assertEquals(1, newOfficers.size());
assertEquals(TEST_EMAIL, newOfficers.get(0));
....

Same for below.